### ISSUE TITLE
Can someone confirm if my understanding of the returned code is valid?

### ISSUE THREAD

**Issue by khrome83** [squizlabs/HTML_CodeSniffer#211]
From a structure standpoint,  what are the different parts of these examples?

`WCAG2AA.Principle1.Guideline1_3.1_3_1.H49.AlignAttr`
`WCAG2AA.Principle1.Guideline1_4.1_4_3_F24.F24.FGColour`
`WCAG2AA.Principle2.Guideline2_4.2_4_4.H77,H78,H79,H80,H81`

I understand the first part is the standard ran against.

https://www.w3.org/TR/WCAG20/#guidelines

Looking at the second example above - 

I understand that the `Principal1` identifies the section of the guidelines that talks about *Perceivable*.  After that I am lost. There is no`1.4.1` with a `4_3` listed with it (2nd example).This is assuming `Guideline1_` is just `1.x` in the `Perceivable` section.  I don't know how to make this back to the guidelines.  I don't know what the underscore separated numbers are after 1_4.1. I don't know where the F24 is in the second example, as they are not listed here:

https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html

Thanks!

**Comment by ironikart** [comment]
I wasn't involved personally in the early days of this project where that would have been laid down, but the structure you are seeing there is used as an internal namespace for 'sniffs' which is the name we give to the individual accessibility checks in HTMLCS. I'll break it into parts and give my understanding of each one. Underscores are used to make the string machine readable.

1. WCAG2AA - The standard being checked against
2. Principle1  - Number of the principle being checked against (1. Perceivable, 2. Operable, 3. Understandable, 4. Robust)
3. Guideline1_3 - Internal group name for the guideline being checked (means it's grouped into checks against 1.3)
4. 1_3_1 - The specific guideline being checked is 1.1.3
5. H49 - The technique being used (e.g. https://www.w3.org/TR/WCAG20-TECHS/H49.html)
6. AlignAttr - An internal name given to the specific failure (in this example there is a problem with the align attribute).

TL;DR it's a historical namespace for errors that should be unique.

**Comment by khrome83** [comment]
@ironikart  - thank you so much, that really helped!
